### PR TITLE
added parenthesis delimiters for emojis/swapped levenshtein matching …

### PR DESCRIPTION
**Pull request by 10plusY**
…for simple matching (too slow)

addresses https://github.com/braidchat/meta/issues/188

### CODE REVIEW

File: src/chat/client/views/new_message.cljs [hunk 1]
```diff
@@ -64,11 +64,11 @@
   [
    ; ... :emoji  -> autocomplete emoji
    (fn [text thread-id]
-     (let [pattern #"\B:(\S{2,})$"]
+     (let [pattern #"\B[:(](\S{2,})$"]
        (when-let [query (second (re-find pattern text))]
          (->> emoji/unicode
               (filter (fn [[k v]]
-                        (fuzzy-matches? k (str ":" query ":"))))
+                        (simple-matches? k (str query))))
```
Comment (rafd) [inline src/chat/client/views/new_message.cljs]: `str` is now redundant here

File: src/chat/client/views/new_message.cljs [hunk 2]
```diff
@@ -77,7 +77,10 @@
                         (string/replace text pattern (str k " ")))
                       :html
                       (fn []
-                        (om/build emoji-view k {:react-key k}))}))))))
+                        (om/build emoji-view (if (= "(" (first text))
+                                                 (str "(" (apply str (rest (butlast k))) ")")
+                                                  k)
+                                  {:react-key k}))}))))))
```
Comment (rafd) [inline src/chat/client/views/new_message.cljs]: you can communicate the intent better by:
```
(let [show-brackets? (= (first text) "(")
       emoji-name (apply str (-> k rest butlast))]
 ....
        
```

File: src/chat/client/views/new_message.cljs [hunk 3]
```diff
@@ -77,9 +77,11 @@
                         (string/replace text pattern (str k " ")))
                       :html
                       (fn []
-                        (om/build emoji-view (if (= "(" (first text))
-                                                 (str "(" (apply str (rest (butlast k))) ")")
-                                                  k)
+                        (om/build emoji-view (let [show-brackets? (= "(" (first text))
+                                                   emoji-name (apply str (-> k rest butlast))]
+                                               (if (true? show-brackets?)
+                                                 (str "(" emoji-name ")")
```
Comment (rafd) [inline src/chat/client/views/new_message.cljs]: true? is unnecessary here b/c show-brackets? returns a boolean

`(if show-brackets? ... ` suffices

**Comment by 10plusY** [comment]
Ahhhhhh sorry

File: src/chat/client/views/new_message.cljs (hunk 3 above)
Comment (10plusY) [inline src/chat/client/views/new_message.cljs]: ahh sorry

**Pull request merged**